src/braket/aws/aws_device.py

```diff
@@ -622,3 +649,111 @@ def get_device_region(device_arn: str) -> str:
+                        json.loads(f.read().decode("utf-8"))
+                    )
+                    return GateCalibrations(json_calibration_data)
+            except urllib.error.URLError:
+                raise urllib.error.URLError(
+                    f"Unable to reach {self.properties.pulse.nativeGateCalibrationsRef}"
+                )
+        else:
+            return None
+
+    def _parse_waveforms(self, waveforms_json: Dict) -> Dict:
+        waveforms = dict()
+        for waveform in waveforms_json:
+            parsed_waveform = _parse_waveform_from_json(waveforms_json[waveform])
+            waveforms[parsed_waveform.id] = parsed_waveform
+        return waveforms
+
+    def _parse_pulse_sequence(
+        self, calibration: Dict, waveforms: Dict[ArbitraryWaveform]
+    ) -> PulseSequence:
+        calibration_sequence = PulseSequence()
+        for instr in calibration:
+            calibration_sequence = calibration_sequence._parse_json(instr, waveforms, self.frames)
+        return calibration_sequence
+
+    def _parse_calibration_json(
+        self, calibration_data: Dict
+    ) -> Dict[Tuple[Gate, QubitSet], PulseSequence]:
+        """
+        Takes the json string from the device calibration URL and returns a structured dictionary of
+        corresponding `Dict[Tuple[Gate, QubitSet], PulseSequence]` to represent the calibration data.
+
+        Args:
+            calibration_data (Dict): The data to be parsed. Based on
+                https://github.com/aws/amazon-braket-schemas-python/blob/main/src/braket/device_schema/pulse/native_gate_calibrations_v1.py.
+
+        Returns:
+            Dict[Tuple[Gate, QubitSet], PulseSequence]: The
+            structured data based on a mapping of `Tuple[Gate, Qubit]` to its calibration repesented as a
+            `PulseSequence`.
+
+        """  # noqa: E501
+        waveforms = self._parse_waveforms(calibration_data["waveforms"])
+        parsed_calibration_data = {}
+        for qubit_node in calibration_data["gates"]:
+            qubit = calibration_data["gates"][qubit_node]
+            for gate_node in qubit:
+                for gate in qubit[gate_node]:
+                    gate_obj = Gate._str_to_gate(gate_node.capitalize())
+                    qubits = QubitSet([int(x) for x in gate["qubits"]])
+                    if gate_obj is None:
+                        # We drop out gates that are not implemented in the BDK
+                        continue
+
+                    argument = None
+                    if len(gate["arguments"]):
+                        argument = (
+                            float(gate["arguments"][0])
+                            if is_float(gate["arguments"][0])
+                            else FreeParameter(gate["arguments"][0])
+                        )
+                    gate_qubit_key = (
+                        (gate_obj(argument), qubits) if argument else (gate_obj(), qubits)
```

Comment:
Why does the service not specify an argument name? What does the service return format actually align with? Align with the BDK? Align with JAQCD? Align with OpenQASM?